clusterloader2/pkg/measurement/util/pvc_store.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This file is copy of https://github.com/kubernetes/kubernetes/blob/master/test/utils/pod_store.go
+with slight changes regarding labelSelector and flagSelector applied and changing to PVCs.
+*/
+
+package util
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+// PVCStore is a convenient wrapper around cache.Store that returns list of v1.PVC instead of interface{}.
+type PVCStore struct {
+	cache.Store
+	stopCh    chan struct{}
+	Reflector *cache.Reflector
+}
+
+// NewPVCStore creates PVCStore based on given namespace, label selector and field selector.
+func NewPVCStore(c clientset.Interface, selector *ObjectSelector) (*PVCStore, error) {
+	lw := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			options.LabelSelector = selector.LabelSelector
+			options.FieldSelector = selector.FieldSelector
+			obj, err := c.CoreV1().PersistentVolumeClaims(selector.Namespace).List(options)
+			return runtime.Object(obj), err
```

Comment:
Why is this cast needed?

Reply:
I don't think  it is. Happy to change it in this PR or open a new one since the PodStore is done this way too.

Reply:
It's returning *v1.PodList:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/kubernetes/typed/core/v1/pod.go#L46

I'm think that it implements runtime.Object interface (otherwise a bunch of other things won't work).

Reply:
And we use it in a bunch of places without casting, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/secret/secret_manager.go#L140

so it has to work.

Reply:
ack, just changing for pvc_store then